iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -92,13 +97,56 @@ public VariableDeclaration VARIABLE_DECLARATION() {
         b.token(EOL)));
   }
 
+  public ResourceDeclaration RESOURCE_DECLARATION() {
+    return b.<ResourceDeclaration>nonterminal(BicepLexicalGrammar.RESOURCE_DECLARATION).is(
+      f.resourceDeclaration(
+        b.token(BicepKeyword.RESOURCE),
+        IDENTIFIER(),
+        INTERPOLATED_STRING_TYPE(),
+        b.optional(b.token(BicepKeyword.EXISTING)),
+        b.token(Punctuator.EQU),
+        OBJECT_EXPRESSION(),
+        b.token(BicepLexicalGrammar.EOL)));
+  }
+
+  public InterpolatedString INTERPOLATED_STRING_TYPE() {
+    return b.<InterpolatedString>nonterminal(BicepLexicalGrammar.INTERPOLATED_STRING).is(
+      f.interpolatedString(
+        b.token(BicepLexicalGrammar.OPENING_APOSTROPHE),
+        b.token(BicepLexicalGrammar.QUOTED_STRING_LITERAL),
+        b.token(Punctuator.APOSTROPHE)));
+  }
+
+  // object -> "{" ( NL+ ( property NL+ )* )? "}"
+  public ObjectExpression OBJECT_EXPRESSION() {
+    return b.<ObjectExpression>nonterminal(BicepLexicalGrammar.OBJECT_EXPRESSION).is(
+      f.objectExpression(
+        b.token(Punctuator.LCURLYBRACE),
+        b.optional(PROPERTY_LIST()),
```

Comment:
```suggestion
        b.zeroOrMore(PROPERTY()),
```
IMO there is no need for a PROPERTY_LIST rule because no tree is created here and you do not have boilerplate code in the TreeFactory.